src/gp_init.c

```diff
@@ -255,15 +256,21 @@ int init_userproxy_socket(struct gssproxy_ctx *gpctx)
 static void hup_handler(verto_ctx *vctx UNUSED, verto_ev *ev)
 {
     int ret;
-    struct gssproxy_ctx *gpctx;
+    struct gssproxy_ctx *gpctx = verto_get_private(ev);
     struct gp_config *new_config, *old_config;
 
-    gpctx = verto_get_private(ev);
+    sd_notifyf(0, "RELOADING=1\n"
+               "MONOTONIC_USEC=%" PRIu64 "\n"
+               "STATUS=Reloading configuration\n",
+               time_now_usec());
 
     GPDEBUG("Received SIGHUP; re-reading config.\n");
     new_config = read_config(gpctx->config_file, gpctx->config_dir,
                              gpctx->config_socket, gpctx->daemonize);
     if (!new_config) {
+        sd_notifyf(0, "READY=1\n"
+                   "STATUS=Running, %i service(s) configured\n",
+                   gpctx->config->num_svcs);
```

Comment:
Shouldn't we notify of an error status/warning of some kind, or does sd_notify not deal with thst ?

Reply:
Maybe just add "(Failed to re-read config)" in the STATUS part?

Reply:
Good idea, will fix

Reply:
> Shouldn't we notify of an error status/warning of some kind, or does sd_notify not deal with thst ?

I don't think it's necessary to check the return value of `sd_notify` because there's not much that can be done about it....if `sd_notify` doesn't work, the system is either truly borked (and `sd_notifyf` is the least of your problems) or there's code somewhere that is unsetting the relevant environment variable. The man page for `sd_notifyf` says "`it is generally recommended to ignore the return value of this call`"